scripts/common/rook.sh

```diff
@@ -173,3 +173,238 @@ function maybe_cleanup_rook() {
 function rook_osd_phase_ready() {
     [ "$(kubectl -n rook-ceph get cephcluster rook-ceph --template '{{.status.phase}}')" = 'Ready' ]
 }
+
+function current_rook_version() {
+    kubectl -n rook-ceph get deploy rook-ceph-operator -oyaml 2>/dev/null \
+        | grep ' image: ' \
+        | awk -F':' 'NR==1 { print $3 }' \
+        | sed 's/v\([^-]*\).*/\1/'
+}
+
+# checks if rook should be upgraded before upgrading k8s. If it should, reports that as an addon, and starts the upgrade process.
+function report_upgrade_rook() {
+    if should_upgrade_rook_10_to_14; then
+        ROOK_10_TO_14_VERSION="v1.0.0" # if you change this code, change the version
+        report_addon_start "rook_10_to_14" "$ROOK_10_TO_14_VERSION"
+        export REPORTING_CONTEXT_INFO="rook_10_to_14 $ROOK_10_TO_14_VERSION"
+        rook_10_to_14
+        export REPORTING_CONTEXT_INFO=""
+        report_addon_success "rook_10_to_14" "$ROOK_10_TO_14_VERSION"
+    fi
+}
+
+# checks the currently installed rook version and the desired rook version
+# if the current version is 1.0-3.x and the desired version is 1.4.9+, returns true
+function should_upgrade_rook_10_to_14() {
+    # rook is not currently installed, so no upgrade
+    if ! is_rook_1 ; then
+        return 1
+    fi
+
+    # rook is not requested to be installed, so no upgrade
+    if [ -z "${ROOK_VERSION}" ]; then
+        return 1
+    fi
+
+    current_version="$(current_rook_version)"
+    semverParse "${current_version}"
+    current_rook_version_major="${major}"
+    current_rook_version_minor="${minor}"
+
+    semverParse "${ROOK_VERSION}"
+    next_rook_version_major="${major}"
+    next_rook_version_minor="${minor}"
+    next_rook_version_patch="${patch}"
+
+    # rook 1.0 currently running
+    if [ "$current_rook_version_major" -eq "1" ] && [ "$current_rook_version_minor" -eq "0" ]; then
+           # rook 1.4+ desired
+            if [ "$next_rook_version_major" -eq "1" ] && [ "$next_rook_version_minor" -ge "4" ] && [ "$next_rook_version_patch" -ge "9" ]; then
```

Comment:
why are you targeting patch version .9+ here?

Reply:
because no one should be updating to 1.4.3, essentially
and so this lets me install 1.4.9 instead of 1.4.3

Reply:
want to change the comment? `rook 1.4.9+ desired`